crates/ty_python_semantic/src/program.rs

```diff
@@ -56,9 +60,11 @@ impl Program {
             self.set_python_platform(db).to(python_platform);
         }
 
-        if python_version != self.python_version(db) {
-            tracing::debug!("Updating python version: Python {python_version}");
-            self.set_python_version(db).to(python_version);
+        if let Some(python_version) = python_version {
+            if python_version != self.python_version(db) {
+                tracing::debug!("Updating python version: Python {python_version}");
+                self.set_python_version(db).to(python_version);
+            }
         }
```

Comment:
I think this is wrong but you have to test it by running ty in `--watch` and then change from having a `python-version` in the `pyproject.toml` to removing it. I suspect that ty will not propagate the change and instead keeps using the old `python-version`.

What we need here is the same behavior as when initializing the settings the first time.

Reply:
Oh interesting. I can look into that.